main.c

```diff
@@ -1295,8 +1296,22 @@ int main(int argc, char **argv) {
 		create_surface(surface);
 	}
 
+	if (state.ext_session_lock_manager_v1) {
+		while (!state.locked) {
```

Comment:
Hmm, if we do not enter the locked state for whatever reason, this turns into an infinite loop. I wonder if we should have a timeout here...

Reply:
We get a `finished` event in that case, and exit.

Reply:
Yeah, fair enough, we already set up the lock and no need to code around a broken compositor.